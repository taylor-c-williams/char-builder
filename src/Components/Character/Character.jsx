export default function Character({ head, middle, pants }) {
	return (
		<section className="character-display">
			{/* Head Display */}
			<article
				aria-label="head"
				style={{
					backgroundImage: `url(./${head}-head.png)`,
					height: 150,
					width: 150,
					backgroundSize: 'cover',
					backgroundRepeat: 'no-repeat',
				}}
			></article>

			{/* Middle Display */}
			<article
				aria-label="middle"
				style={{
					backgroundImage: `url(./${middle}-middle.png)`,
					height: 150,
					width: 150,
					backgroundSize: 'cover',
					backgroundRepeat: 'no-repeat',
				}}
			></article>

			{/* Pants Display */}
			<article
				aria-label="pants"
				style={{
					backgroundImage: `url(./${pants}-pants.png)`,
					height: 350,
					width: 150,
					backgroundSize: 'cover',
					backgroundRepeat: 'no-repeat',
				}}
			></article>
		</section>
	);
}
