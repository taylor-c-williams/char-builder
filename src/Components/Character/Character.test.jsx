import { screen, render } from '@testing-library/react';
import Character from './Character';

it('Should render Character component', () => {
	const { container } = render(<Character head="dog" middle="fancy" />);
	const head = screen.getByLabelText(/head/);

	expect(head).toHaveStyle(`background-image: url(./dog-head.png`);
	expect(container).toMatchSnapshot();
});
