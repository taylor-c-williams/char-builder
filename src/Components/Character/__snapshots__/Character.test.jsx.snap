// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Character component 1`] = `
<div>
  <section
    class="character-display"
  >
    <article
      aria-label="head"
      style="background-image: url(./dog-head.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="middle"
      style="background-image: url(./fancy-middle.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="pants"
      style="background-image: url(./undefined-pants.png); height: 350px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
  </section>
</div>
`;
